src/main/java/net/minestom/server/entity/type/projectile/AbstractProjectile.java

```diff
@@ -0,0 +1,25 @@
+package net.minestom.server.entity.type.projectile;
+
+import net.minestom.server.entity.Entity;
+import net.minestom.server.entity.EntityType;
+import net.minestom.server.entity.ObjectEntity;
+import net.minestom.server.entity.type.Projectile;
+import net.minestom.server.utils.Position;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public abstract class AbstractProjectile extends ObjectEntity implements Projectile {
```

Comment:
All projectiles seem to be `ObjectEntities`, so I guess it's ok to force them to inherit it.